hooks/pre-push

```diff
@@ -1,2 +1,13 @@
 #!/bin/bash -e
-for f in $0.d/*; do $f; done
+rust=0
+for file in $(git diff-index --name-only HEAD); do
+    if [[ -n "$TRAVIS" ]] || [[ $file == *.rs ]] || [[ $file == *.toml ]]; then
+        rust=1
+        break
+    fi
+done
+for file in $0.d/*; do
+    if [[ $(basename $file) != cargo-* ]] || [[ $rust == 1 ]]; then
```

Comment:
The obvious way to filter here that I didn't think of before is to put Rust tests in a separate directory.

Reply:
Yep, that also makes for much cleaner code. I've moved things around in the latest push.